polyply/src/topology.py

```diff
@@ -388,8 +388,11 @@ def add_positions_from_file(self, path, skip_res=[], resolution='mol'):
         for meta_mol in self.molecules:
             for meta_node in meta_mol.nodes:
                 resname = meta_mol.nodes[meta_node]["resname"]
-                # the fragment graph nodes are not sorted
-                mol_nodes = sorted(meta_mol.nodes[meta_node]['graph'].nodes)
+                # the fragment graph nodes are not sorted so we sort them by index
+                # as defined in the itp-file to capture cases, where the molecule
+                # graph nodes are permuted with respect to the index
+                idx_nodes = nx.get_node_attributes(meta_mol.nodes[meta_node]['graph'], "index")
+                mol_nodes = [node for _, node in sorted(zip(idx_nodes.values(), idx_nodes.keys()))]
```

Comment:
@pckroon I don't get it but obviously this doesn't work.

Reply:
Because I'm an idiot (:

```py
idx_nodes = nx.get_node_attributes(meta_mol.nodes[meta_node]['graph'], "index")
mol_nodes = sorted(meta_mol.nodes[meta_node]['graph'], key=idx_nodes.get)
```
should do it

Reply:
Or `mol_nodes = sorted(idx_nodes, key=idx_nodes.get)
`